In [1]:
%load_ext sql

In [3]:
con="postgresql://postgres:macbook@127.0.0.1/pagila"
%sql $con

'Connected: postgres@pagila'

<img src='files/facts.png'>

## Roll-up
- Stepping up the level of aggregation to a large grouping
- e.g.`city` is summed as `country`

TODO: Write a query that calculates revenue (sales_amount) by day, rating, and country. Sort the data by revenue in descending order, and limit the data to the top 20 results. The first few rows of your output should match the table below.

In [5]:
%%sql select d.day,m.rating,c.country,sum(sales_amount) as re
from factsales as f
join dimdate as d on d.date_key=f.date_key
join dimmovie as m on m.movie_key=f.movie_key
join dimcustomer as c on c.customer_key=f.customer_key
group by(c.country,d.day,m.rating)
order by re desc
limit 10

 * postgresql://postgres:***@127.0.0.1/pagila
10 rows affected.


[(30, 'G', 'China', Decimal('169.67')),
 (30, 'PG', 'India', Decimal('156.67')),
 (30, 'NC-17', 'India', Decimal('153.64')),
 (30, 'PG-13', 'China', Decimal('146.67')),
 (30, 'R', 'China', Decimal('145.66')),
 (30, 'R', 'India', Decimal('143.68')),
 (30, 'G', 'India', Decimal('137.67')),
 (18, 'NC-17', 'India', Decimal('135.75')),
 (30, 'PG', 'China', Decimal('131.72')),
 (21, 'PG-13', 'India', Decimal('128.74'))]

<div class="p-Widget jp-RenderedHTMLCommon jp-RenderedHTML jp-mod-trusted jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html"><table>
    <tbody><tr>
        <th>day</th>
        <th>rating</th>
        <th>country</th>
        <th>revenue</th>
    </tr>
    <tr>
        <td>30</td>
        <td>G</td>
        <td>China</td>
        <td>169.67</td>
    </tr>
    <tr>
        <td>30</td>
        <td>PG</td>
        <td>India</td>
        <td>156.67</td>
    </tr>
    <tr>
        <td>30</td>
        <td>NC-17</td>
        <td>India</td>
        <td>153.64</td>
    </tr>
    <tr>
        <td>30</td>
        <td>PG-13</td>
        <td>China</td>
        <td>146.67</td>
    </tr>
    <tr>
        <td>30</td>
        <td>R</td>
        <td>China</td>
        <td>145.66</td>
    </tr>
</tbody></table></div>

## Drill-down
- Breaking up one of the dimensions to a lower level.
- e.g.`city` is broken up into  `districts`

TODO: Write a query that calculates revenue (sales_amount) by day, rating, and district. Sort the data by revenue in descending order, and limit the data to the top 20 results. The first few rows of your output should match the table below.

In [6]:
%%sql select d.day,m.rating,c.district,sum(sales_amount) as re
from factsales as f
join dimdate as d on d.date_key=f.date_key
join dimmovie as m on m.movie_key=f.movie_key
join dimcustomer as c on c.customer_key=f.customer_key
group by(c.district,d.day,m.rating)
order by re desc
limit 10

 * postgresql://postgres:***@127.0.0.1/pagila
10 rows affected.


[(30, 'PG-13', 'Southern Tagalog', Decimal('53.88')),
 (30, 'G', 'Inner Mongolia', Decimal('38.93')),
 (30, 'G', 'Shandong', Decimal('36.93')),
 (30, 'NC-17', 'West Bengali', Decimal('36.92')),
 (17, 'PG-13', 'Shandong', Decimal('34.95')),
 (1, 'PG', 'California', Decimal('32.94')),
 (18, 'NC-17', 'So Paulo', Decimal('32.93')),
 (21, 'R', 'So Paulo', Decimal('31.93')),
 (30, 'NC-17', 'Buenos Aires', Decimal('31.93')),
 (30, 'PG', 'Southern Tagalog', Decimal('30.94'))]

<div class="p-Widget jp-RenderedHTMLCommon jp-RenderedHTML jp-mod-trusted jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html"><table>
    <tbody><tr>
        <th>day</th>
        <th>rating</th>
        <th>district</th>
        <th>revenue</th>
    </tr>
    <tr>
        <td>30</td>
        <td>PG-13</td>
        <td>Southern Tagalog</td>
        <td>53.88</td>
    </tr>
    <tr>
        <td>30</td>
        <td>G</td>
        <td>Inner Mongolia</td>
        <td>38.93</td>
    </tr>
    <tr>
        <td>30</td>
        <td>G</td>
        <td>Shandong</td>
        <td>36.93</td>
    </tr>
    <tr>
        <td>30</td>
        <td>NC-17</td>
        <td>West Bengali</td>
        <td>36.92</td>
    </tr>
    <tr>
        <td>17</td>
        <td>PG-13</td>
        <td>Shandong</td>
        <td>34.95</td>
    </tr>
</tbody></table></div>